### PR TITLE
PRP-177

### PR DESCRIPTION
**Pull request by microwavenby**
## Ticket

https://wicmtdp.atlassian.net/browse/PRP-177

## Changes
* Created new components for collecting names, adjunctive eligibility, dates of birth
* Created index, intro, count and details page routes
* Added all new pages and components to Storybook
* Created unit tests for components

### CODE REVIEW

File: .github/workflows/cd-chromatic.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# .github/workflows/chromatic.yml
+
+# Workflow name
+name: '🌈 Chromatic'
+
+# Event for the workflow
+on: pull_request
```
Comment (rocketnova) [inline .github/workflows/cd-chromatic.yml]: Let's only trigger Chromatic builds for changes to the participant portal
```suggestion
on: 
  pull_request:
        paths:
          - participant_portal/**
```

File: participant_portal/package.json [hunk 2]
```diff
@@ -112,5 +113,8 @@
   },
   "prisma": {
     "seed": "ts-node  --require tsconfig-paths/register --compiler-options {\"module\":\"CommonJS\"} prisma/seed.ts"
-  }
+  },
+  "version": "",
+  "readme": "ERROR: No README data found!",
+  "_id": "participant_portal@"
```
Comment (rocketnova) [inline participant_portal/package.json:119]: Curious: what do these do?

File: participant_portal/public/locales/en/common.json [hunk 3]
```diff
@@ -1,34 +1,92 @@
 {
+  "AdjunctiveEligibility": {
+    "label": "Are they enrolled in any of these programs?",
+    "programs": [
+      "Medicaid / Healthy Montana Kids Plus",
+      "SNAP (Supplemental Nutrition Assistance Program",
```
Comment (rocketnova) [inline participant_portal/public/locales/en/common.json]:
```suggestion
      "SNAP (Supplemental Nutrition Assistance Program)",
```

File: participant_portal/public/locales/en/common.json [hunk 4]
```diff
@@ -1,34 +1,92 @@
...
+      "TANF (Temporary Assistance for Needy Families)",
+      "FDPIR (Food Distribution Program on Indian Reservations"
```
Comment (rocketnova) [inline participant_portal/public/locales/en/common.json]:
```suggestion
      "FDPIR (Food Distribution Program on Indian Reservations)"
```

File: participant_portal/stories/components/AdjunctiveInput.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  title: "Components/Input/AdjunctiveInput",
+};
+
+const AdjunctiveTemplate = {
+  render: (props: AdjunctiveInputProps) => {
+    return <AdjunctiveInput {...props} />;
+  },
+};
+
+const defaultAdjunctiveProps: Omit<AdjunctiveInputProps, "name"> = {
```
Comment (rocketnova) [inline participant_portal/stories/components/AdjunctiveInput.stories.tsx:15]: Curious: how do these `Omit` types work?

File: participant_portal/stories/components/Relationship.stories.tsx
Comment (rocketnova) [inline participant_portal/stories/components/Relationship.stories.tsx:1]: Let's rename this to `RelationshipInput.stories.tsx` to match the component.

File: participant_portal/tests/components/RelationshipInput.test.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+const testProps: RelationshipInputProps = {
+  relationshipKey: "test:relationship",
+  required: false,
+  legendKey: "test:relationship.label",
+  legendStyle: "default",
+  name: "relationship",
+};
+
+it("renders the default adjunctive eligibility field", () => {
```
Comment (rocketnova) [inline participant_portal/tests/components/RelationshipInput.test.tsx:15]: This test name seems wrong?

File: participant_portal/app/components/ParticipantCard.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+          relationshipKey={relationshipKey}
+          legendKey={`${relationshipKey}.label`}
+          name={`participant-${index}-relationship`}
+        />
+        <NameInput {...nameProps} />
+        <DateInput {...dateProps} />
+        <AdjunctiveInput {...adjunctiveProps} />
+      </CardBody>
+      <CardFooter>
+        {" "}
```
Comment (rocketnova) [inline participant_portal/app/components/ParticipantCard.tsx]: Is this to enter a blank line? If so, can we accomplish this with css classes instead?

File: participant_portal/app/routes/details.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        <Trans i18nKey="Detail.body" />
+      </div>
+      <Form>
+        <CardGroup>{participantCards}</CardGroup>
+        <Button
+          className="display-block margin-top-6"
+          type="submit"
+          formMethod="post"
+        >
+          {t("Detail.button")}
```
Comment (rocketnova) [inline participant_portal/app/routes/details.tsx:65]: I see that you haven't implemented the add card/remove card logic to this PR. Ok if I add it to the acceptance criteria of https://wicmtdp.atlassian.net/browse/PRP-204?

**Review by rocketnova — APPROVED** [review]
This is an incredible amount of work. Thank you so much! Looks great. Small comments and questions inline.

File: participant_portal/package.json (hunk 2 above)
Comment (microwavenby) [reply participant_portal/package.json:119]: No idea. I think chromatic's setup process put them in?

File: participant_portal/stories/components/AdjunctiveInput.stories.tsx (hunk 5 above)
Comment (microwavenby) [reply participant_portal/stories/components/AdjunctiveInput.stories.tsx:15]: https://timmousk.com/blog/typescript-utility-types/

Omit is a Typescript helper for type partials, it "omits" whatever key (or multiple | keys) you pass it from the type

File: participant_portal/app/routes/details.tsx (hunk 8 above)
Comment (microwavenby) [reply participant_portal/app/routes/details.tsx:65]: Yep, on both accounts

**Pull request merged**